packages/runtime-common/realm.ts

```diff
@@ -298,8 +301,17 @@ export class Realm {
     let loader = virtualNetwork.createLoader();
     adapter.setLoader?.(loader);
 
+    this.#realmAuthHandler = new RealmAuthHandler(
+      this.#matrixClient,
+      loader,
+      url,
+    );
     this.loaderTemplate = loader;
     this.loaderTemplate.registerURLHandler(this.maybeHandle.bind(this));
+    this.loaderTemplate.registerURLHandler(
+      this.#realmAuthHandler.fetchWithAuth,
```

Comment:
This handler will add authorization header to the request that the realm makes, if needed